src/main/java/net/coderbot/iris/uniforms/transforms/SmoothedFloat.java

```diff
@@ -38,25 +38,23 @@
 	 */
 	private boolean hasInitialValue;
 
+	private float halfLifeUp;
+
+	private float halfLifeDown;
+
 	/**
 	 * Creates a new SmoothedFloat with a given half life.
 	 *
-	 * @param halfLife   the half life in the exponential decay, in deciseconds (1/10th of a second) / 2 ticks.
+	 * @param halfLifeUp   the half life in the exponential decay, in deciseconds (1/10th of a second) / 2 ticks.
 	 *                   For example, a half life of value of 2.0 is 4 ticks or 0.2 seconds
 	 * @param unsmoothed the input sequence of unsmoothed values to be smoothed. {@code unsmoothed.getAsFloat()} will be
 	 *                   called exactly once for every time {@code smoothed.getAsFloat()} is called.
 	 */
-	public SmoothedFloat(float halfLife, FloatSupplier unsmoothed, FrameUpdateNotifier updateNotifier) {
+	public SmoothedFloat(float halfLifeUp, float halfLifeDown, FloatSupplier unsmoothed, FrameUpdateNotifier updateNotifier) {
 		// Half life is measured in units of 10ths of a second, or 2 ticks
 		// For example, a half life of value of 2.0 is 4 ticks or 0.2 seconds
-		halfLife *= 0.1f;
-
-		// Compute the decay constant from the half life
-		// https://en.wikipedia.org/wiki/Exponential_decay#Measuring_rates_of_decay
-		// https://en.wikipedia.org/wiki/Exponential_smoothing#Time_constant
-		double timeConstant = halfLife / LN_OF_2;
-		// k = 1 / τ
-		this.decayConstant = (float) (1.0f / timeConstant);
+		this.halfLifeUp = (halfLifeUp * 0.1F);
```

Comment:
Why not compute and store two decay constants? Could move the decay constant computation into a function to avoid duplication

Reply:
Will do

Reply:
Done